src/ch05-01-language-basics.md

```diff
@@ -112,6 +118,31 @@ let expressions work similarly to how they work in Haskell.
     sha256 = "...";
   };
 ```
+
+## Inherit statements
+
+An inherit statement can be used to _pull_ values out of a parent scope
+into the current one. Values are whitespace separated and the statement
+ends with a `;`. They are valid only in `let` blocks and inside
+attribute sets.
+
+If an attribute set in parenthesis follows immediately after the `inherit`
+keyword, then values can also be pulled directly out of the given set.
+
+In a `let` block, inherit statements can also come before the values they
+reference, so long as they are in the same block.
+
+```nix
+let
+  inherit (set.values) name version;
+
+  set.values = {
+    name = "package";
+    version = "0.1.0";
+  };
+in
+{ inherit name version; } == set.values
+```
```

Comment:
I was deliberately attempting to be concise, although I'm not super against splitting it in two example, however in this case isn't the `let` block part of the demonstration since it's one of two places where an `inherit` can ocur?

Also, in regards to repl style examples, I was attempting to follow the Rust example of writing valid code blocks that could be pulled out and evaluated independantly to a boolean. It'd be great if we could somehow find a way to automate this to ensure our examples stay valid.

However, maybe this isn't as important as with Rust since Nix doesn't really change that much?

Reply:
I think there's a lot going on in the blocks, let ... in block, recursive definition of `set`, then inheriting the value, and then a comparison which isn't demonstrated.

The rust-lang book has a leg up in that you can just execute code. Until then we will need to be more acommodating.

The other thing is that it's not apparent how to run the code, `nix eval -f example.nix` would probably be right, but that's not apparent to the user. 

I've been doing
```
$ cat <file>

$ <nix command> <file>
```

For now, but it would be nice to move away from this. Just not sure what would be best.

Reply:
Perhaps what is needed is to develop a mdbook plugin to actually evaluate the code in browser _and_ CI, similar to Rust. This is definitely outside of the scope of this PR though.

I went ahead and split this into two (hopefully simpler) examples.